cmd/registry/controller/parser.go

```diff
@@ -69,8 +70,37 @@ func ExtendSourcePattern(
 
 }
 
+func ResourceNameFromDependency(
```

Comment:
That file is in the same package as this one. If that's the only place this is called I think we can unexport this for now. As a rule, you can know if you're calling a function from a different package because the package's name will need to be included, e.g. `regexp.MustCompile` is in the `regexp` package.